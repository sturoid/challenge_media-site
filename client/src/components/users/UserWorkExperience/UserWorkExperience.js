import React from 'react';
import PropTypes from 'prop-types';
import Table from '../../_shared/Table';

const UserWorkExperience = ({ work }) => (
  <Table>
    <thead>
      <tr>
        <th>Position</th>
        <th>Organization</th>
        <th>Location</th>
      </tr>
    </thead>
    <tbody>
      {work.map(w => (
        <tr key={w.position + w.organization + w.location}>
          <td>{w.position}</td>
          <td>{w.organization}</td>
          <td>{w.location}</td>
        </tr>
      ))}
    </tbody>
  </Table>
);

UserWorkExperience.propTypes = {
  work: PropTypes.arrayOf(
    PropTypes.shape({
      position: PropTypes.string.isRequired,
      organization: PropTypes.string.isRequired,
      location: PropTypes.string.isRequired
    })
  ).isRequired
};

export default UserWorkExperience;
