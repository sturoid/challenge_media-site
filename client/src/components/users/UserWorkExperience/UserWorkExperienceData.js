import React from 'react';
import PropTypes from 'prop-types';
import axios from 'axios';
import Loader from '../../../components/_shared/Loader';
import UserWorkExperience from './UserWorkExperience';

class UserWorkExperienceData extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      loading: false,
      error: false,
      work: []
    };
  }

  componentDidMount() {
    this.fetchDataAndSetState();
  }

  async fetchDataAndSetState() {
    this.setState({ loading: true });

    const { data: work } = await axios.get(
      `/api/v1/user/${this.props.userId}/work-experience`
    );

    this.setState({ work, loading: false });
  }

  render() {
    const { loading, error, work } = this.state;
    if (loading) return <Loader loading={loading} />;
    if (error) return <div>{error.message}</div>;
    return <UserWorkExperience work={work} />;
  }
}

UserWorkExperienceData.propTypes = {
  userId: PropTypes.string.isRequired
};

export default UserWorkExperienceData;
