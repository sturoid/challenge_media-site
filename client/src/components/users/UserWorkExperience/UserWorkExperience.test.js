import React from 'react';
import renderer from 'react-test-renderer';
import UserWorkExperience from './UserWorkExperience';
import workExperienceData from '../../../../../lib/test-data/userWorkExperience';

describe('<UserWorkExperience />', () => {
  it('renders correctly and equals snapshot', () => {
    const tree = renderer
      .create(<UserWorkExperience work={workExperienceData} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
