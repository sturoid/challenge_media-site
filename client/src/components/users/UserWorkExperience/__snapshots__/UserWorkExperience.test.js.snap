// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserWorkExperience /> renders correctly and equals snapshot 1`] = `
<div
  className="css-qijqp5"
>
  <table
    className="css-1z0wrym"
  >
    <thead>
      <tr>
        <th>
          Position
        </th>
        <th>
          Organization
        </th>
        <th>
          Location
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          Co-Founder & Chief of Design
        </td>
        <td>
          Behance
        </td>
        <td>
          Brooklyn, NY, USA
        </td>
      </tr>
      <tr>
        <td>
          Senior Designer
        </td>
        <td>
          AR Media
        </td>
        <td>
          Brooklyn, NY, USA
        </td>
      </tr>
      <tr>
        <td>
          Senior Designer
        </td>
        <td>
          Pro Am
        </td>
        <td>
          New York, NY, USA
        </td>
      </tr>
      <tr>
        <td>
          Junior Designer
        </td>
        <td>
          Zeligmedia
        </td>
        <td>
          Barcelona, Spain
        </td>
      </tr>
      <tr>
        <td>
          Junior Designer
        </td>
        <td>
          Triada ComunicaciÃ³n
        </td>
        <td>
          Barcelona, Spain
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
